<<<<<<< HEAD
---
description: 'Enforce valid definition of `new` and `constructor`.'
---

> 🛑 This file is source code, not the primary documentation location! 🛑
>
> See **https://typescript-eslint.io/rules/no-misused-new** for documentation.

JavaScript classes may define a `constructor` method that runs when a class instance is newly created.
TypeScript allows interfaces that describe a static class object to define a `new()` method (though this is rarely used in real world code).
Developers new to JavaScript classes and/or TypeScript interfaces may sometimes confuse when to use `constructor` or `new`.

This rule reports when a class defines a method named `new` or an interface defines a method named `constructor`.

## Examples

<!--tabs-->

### ❌ Incorrect

```ts
declare class C {
=======
# Enforce valid definition of `new` and `constructor` (`no-misused-new`)

Warns on apparent attempts to define constructors for interfaces or `new` for classes.

## Rule Details

Examples of **incorrect** code for this rule.

```ts
class C {
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
  new(): C;
}

interface I {
  new (): I;
  constructor(): void;
}
```

<<<<<<< HEAD
### ✅ Correct

```ts
declare class C {
  constructor();
}

=======
Examples of **correct** code for this rule.

```ts
class C {
  constructor() {}
}
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
interface I {
  new (): C;
}
```

<<<<<<< HEAD
## When Not To Use It

If you intentionally want a class with a `new` method, and you're confident nobody working in your code will mistake it with a constructor.
=======
## Options

```json
{
  "@typescript-eslint/no-misused-new": "error"
}
```

## Compatibility

- TSLint: [no-misused-new](https://palantir.github.io/tslint/rules/no-misused-new/)
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
