<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
---
description: 'Disallow the use of parameter properties in class constructors.'
---

> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/no-parameter-properties** for documentation.

:::danger Deprecated

This rule has been deprecated in favour of the equivalent, better named [`parameter-properties`](./parameter-properties.md) rule.
:::
=======
# Disallow the use of parameter properties in class constructors (`no-parameter-properties`)
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
# Disallow the use of parameter properties in class constructors (`no-parameter-properties`)
=======
=======
>>>>>>> 6be43a322 (figuring out lint)
---
description: 'Disallow the use of parameter properties in class constructors.'
---

> üõë This file is source code, not the primary documentation location! üõë
>
> See **https://typescript-eslint.io/rules/no-parameter-properties** for documentation.

:::danger Deprecated

This rule has been deprecated in favour of the equivalent, better named [`parameter-properties`](./parameter-properties.md) rule.
:::
<<<<<<< HEAD
>>>>>>> 6be43a322 (figuring out lint)
=======
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> 84b8e337c (figuring out lint)

Parameter properties can be confusing to those new to TypeScript as they are less explicit than other ways
of declaring and initializing class members.

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
## Examples
=======
## Rule Details
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
## Rule Details
=======
## Examples
>>>>>>> 6be43a322 (figuring out lint)
<<<<<<< HEAD
>>>>>>> 84b8e337c (figuring out lint)
=======
=======
## Examples
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> ee12506f5 (figuring out lint)

This rule disallows the use of parameter properties in constructors, forcing the user to explicitly
declare all properties in the class.

## Options

This rule, in its default state, does not require any argument and would completely disallow the use of parameter properties.
If you would like to allow certain types of parameter properties then you may pass an object with the following options:

- `allows`, an array containing one or more of the allowed modifiers. Valid values are:
  - `readonly`, allows **readonly** parameter properties.
  - `private`, allows **private** parameter properties.
  - `protected`, allows **protected** parameter properties.
  - `public`, allows **public** parameter properties.
  - `private readonly`, allows **private readonly** parameter properties.
  - `protected readonly`, allows **protected readonly** parameter properties.
  - `public readonly`, allows **public readonly** parameter properties.

### default

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
Examples of code for this rule with no options at all:

<!--tabs-->

#### ‚ùå Incorrect
=======
Examples of **incorrect** code for this rule with no options at all:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **incorrect** code for this rule with no options at all:
=======
=======
>>>>>>> 6be43a322 (figuring out lint)
Examples of code for this rule with no options at all:

<!--tabs-->

#### ‚ùå Incorrect
<<<<<<< HEAD
>>>>>>> 6be43a322 (figuring out lint)
=======
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> 84b8e337c (figuring out lint)

```ts
class Foo {
  constructor(readonly name: string) {}
}

class Foo {
  constructor(private name: string) {}
}

class Foo {
  constructor(protected name: string) {}
}

class Foo {
  constructor(public name: string) {}
}

class Foo {
  constructor(private readonly name: string) {}
}

class Foo {
  constructor(protected readonly name: string) {}
}

class Foo {
  constructor(public readonly name: string) {}
}
```

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
#### ‚úÖ Correct
=======
Examples of **correct** code for this rule with no options at all:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **correct** code for this rule with no options at all:
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
<<<<<<< HEAD
>>>>>>> 84b8e337c (figuring out lint)
=======
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> ee12506f5 (figuring out lint)

```ts
class Foo {
  constructor(name: string) {}
}
```

### readonly

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
Examples of code for the `{ "allows": ["readonly"] }` options:

<!--tabs-->

#### ‚ùå Incorrect
=======
Examples of **incorrect** code for the `{ "allows": ["readonly"] }` options:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **incorrect** code for the `{ "allows": ["readonly"] }` options:
=======
=======
>>>>>>> 6be43a322 (figuring out lint)
Examples of code for the `{ "allows": ["readonly"] }` options:

<!--tabs-->

#### ‚ùå Incorrect
<<<<<<< HEAD
>>>>>>> 6be43a322 (figuring out lint)
=======
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> 84b8e337c (figuring out lint)

```ts
class Foo {
  constructor(private name: string) {}
}

class Foo {
  constructor(protected name: string) {}
}

class Foo {
  constructor(public name: string) {}
}

class Foo {
  constructor(private readonly name: string) {}
}

class Foo {
  constructor(protected readonly name: string) {}
}

class Foo {
  constructor(public readonly name: string) {}
}
```

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
#### ‚úÖ Correct
=======
Examples of **correct** code for the `{ "allows": ["readonly"] }` options:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **correct** code for the `{ "allows": ["readonly"] }` options:
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
<<<<<<< HEAD
>>>>>>> 84b8e337c (figuring out lint)
=======
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> ee12506f5 (figuring out lint)

```ts
class Foo {
  constructor(name: string) {}
}

class Foo {
  constructor(readonly name: string) {}
}
```

### private

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
Examples of code for the `{ "allows": ["private"] }` options:

<!--tabs-->

#### ‚ùå Incorrect
=======
Examples of **incorrect** code for the `{ "allows": ["private"] }` options:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **incorrect** code for the `{ "allows": ["private"] }` options:
=======
=======
>>>>>>> 6be43a322 (figuring out lint)
Examples of code for the `{ "allows": ["private"] }` options:

<!--tabs-->

#### ‚ùå Incorrect
<<<<<<< HEAD
>>>>>>> 6be43a322 (figuring out lint)
=======
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> 84b8e337c (figuring out lint)

```ts
class Foo {
  constructor(readonly name: string) {}
}

class Foo {
  constructor(protected name: string) {}
}

class Foo {
  constructor(public name: string) {}
}

class Foo {
  constructor(private readonly name: string) {}
}

class Foo {
  constructor(protected readonly name: string) {}
}

class Foo {
  constructor(public readonly name: string) {}
}
```

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
#### ‚úÖ Correct
=======
Examples of **correct** code for the `{ "allows": ["private"] }` options:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **correct** code for the `{ "allows": ["private"] }` options:
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
<<<<<<< HEAD
>>>>>>> 84b8e337c (figuring out lint)
=======
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> ee12506f5 (figuring out lint)

```ts
class Foo {
  constructor(name: string) {}
}

class Foo {
  constructor(private name: string) {}
}
```

### protected

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
Examples of code for the `{ "allows": ["protected"] }` options:

<!--tabs-->

#### ‚ùå Incorrect
=======
Examples of **incorrect** code for the `{ "allows": ["protected"] }` options:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **incorrect** code for the `{ "allows": ["protected"] }` options:
=======
=======
>>>>>>> 6be43a322 (figuring out lint)
Examples of code for the `{ "allows": ["protected"] }` options:

<!--tabs-->

#### ‚ùå Incorrect
<<<<<<< HEAD
>>>>>>> 6be43a322 (figuring out lint)
=======
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> 84b8e337c (figuring out lint)

```ts
class Foo {
  constructor(readonly name: string) {}
}

class Foo {
  constructor(private name: string) {}
}

class Foo {
  constructor(public name: string) {}
}

class Foo {
  constructor(private readonly name: string) {}
}

class Foo {
  constructor(protected readonly name: string) {}
}

class Foo {
  constructor(public readonly name: string) {}
}
```

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
#### ‚úÖ Correct
=======
Examples of **correct** code for the `{ "allows": ["protected"] }` options:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **correct** code for the `{ "allows": ["protected"] }` options:
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
<<<<<<< HEAD
>>>>>>> 84b8e337c (figuring out lint)
=======
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> ee12506f5 (figuring out lint)

```ts
class Foo {
  constructor(name: string) {}
}

class Foo {
  constructor(protected name: string) {}
}
```

### public

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
Examples of code for the `{ "allows": ["public"] }` options:

<!--tabs-->

#### ‚ùå Incorrect
=======
Examples of **incorrect** code for the `{ "allows": ["public"] }` options:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **incorrect** code for the `{ "allows": ["public"] }` options:
=======
=======
>>>>>>> 6be43a322 (figuring out lint)
Examples of code for the `{ "allows": ["public"] }` options:

<!--tabs-->

#### ‚ùå Incorrect
<<<<<<< HEAD
>>>>>>> 6be43a322 (figuring out lint)
=======
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> 84b8e337c (figuring out lint)

```ts
class Foo {
  constructor(readonly name: string) {}
}

class Foo {
  constructor(private name: string) {}
}

class Foo {
  constructor(protected name: string) {}
}

class Foo {
  constructor(private readonly name: string) {}
}

class Foo {
  constructor(protected readonly name: string) {}
}

class Foo {
  constructor(public readonly name: string) {}
}
```

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
#### ‚úÖ Correct
=======
Examples of **correct** code for the `{ "allows": ["public"] }` options:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **correct** code for the `{ "allows": ["public"] }` options:
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
<<<<<<< HEAD
>>>>>>> 84b8e337c (figuring out lint)
=======
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> ee12506f5 (figuring out lint)

```ts
class Foo {
  constructor(name: string) {}
}

class Foo {
  constructor(public name: string) {}
}
```

### private readonly

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
Examples of code for the `{ "allows": ["private readonly"] }` options:

<!--tabs-->

#### ‚ùå Incorrect
=======
Examples of **incorrect** code for the `{ "allows": ["private readonly"] }` options:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **incorrect** code for the `{ "allows": ["private readonly"] }` options:
=======
=======
>>>>>>> 6be43a322 (figuring out lint)
Examples of code for the `{ "allows": ["private readonly"] }` options:

<!--tabs-->

#### ‚ùå Incorrect
<<<<<<< HEAD
>>>>>>> 6be43a322 (figuring out lint)
=======
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> 84b8e337c (figuring out lint)

```ts
class Foo {
  constructor(readonly name: string) {}
}

class Foo {
  constructor(private name: string) {}
}

class Foo {
  constructor(protected name: string) {}
}

class Foo {
  constructor(public name: string) {}
}

class Foo {
  constructor(protected readonly name: string) {}
}

class Foo {
  constructor(public readonly name: string) {}
}
```

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
#### ‚úÖ Correct
=======
Examples of **correct** code for the `{ "allows": ["private readonly"] }` options:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **correct** code for the `{ "allows": ["private readonly"] }` options:
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
<<<<<<< HEAD
>>>>>>> 84b8e337c (figuring out lint)
=======
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> ee12506f5 (figuring out lint)

```ts
class Foo {
  constructor(name: string) {}
}

class Foo {
  constructor(private readonly name: string) {}
}
```

### protected readonly

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
Examples of code for the `{ "allows": ["protected readonly"] }` options:

<!--tabs-->

#### ‚ùå Incorrect
=======
Examples of **incorrect** code for the `{ "allows": ["protected readonly"] }` options:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **incorrect** code for the `{ "allows": ["protected readonly"] }` options:
=======
=======
>>>>>>> 6be43a322 (figuring out lint)
Examples of code for the `{ "allows": ["protected readonly"] }` options:

<!--tabs-->

#### ‚ùå Incorrect
<<<<<<< HEAD
>>>>>>> 6be43a322 (figuring out lint)
=======
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> 84b8e337c (figuring out lint)

```ts
class Foo {
  constructor(readonly name: string) {}
}

class Foo {
  constructor(private name: string) {}
}

class Foo {
  constructor(protected name: string) {}
}

class Foo {
  constructor(public name: string) {}
}

class Foo {
  constructor(private readonly name: string) {}
}

class Foo {
  constructor(public readonly name: string) {}
}
```

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
#### ‚úÖ Correct
=======
Examples of **correct** code for the `{ "allows": ["protected readonly"] }` options:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **correct** code for the `{ "allows": ["protected readonly"] }` options:
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
<<<<<<< HEAD
>>>>>>> 84b8e337c (figuring out lint)
=======
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> ee12506f5 (figuring out lint)

```ts
class Foo {
  constructor(name: string) {}
}

class Foo {
  constructor(protected readonly name: string) {}
}
```

### public readonly

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
Examples of code for the `{ "allows": ["public readonly"] }` options:

<!--tabs-->

#### ‚ùå Incorrect
=======
Examples of **incorrect** code for the `{ "allows": ["public readonly"] }` options:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **incorrect** code for the `{ "allows": ["public readonly"] }` options:
=======
=======
>>>>>>> 6be43a322 (figuring out lint)
Examples of code for the `{ "allows": ["public readonly"] }` options:

<!--tabs-->

#### ‚ùå Incorrect
<<<<<<< HEAD
>>>>>>> 6be43a322 (figuring out lint)
=======
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> 84b8e337c (figuring out lint)

```ts
class Foo {
  constructor(readonly name: string) {}
}

class Foo {
  constructor(private name: string) {}
}

class Foo {
  constructor(protected name: string) {}
}

class Foo {
  constructor(public name: string) {}
}

class Foo {
  constructor(private readonly name: string) {}
}

class Foo {
  constructor(protected readonly name: string) {}
}
```

<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
#### ‚úÖ Correct
=======
Examples of **correct** code for the `{ "allows": ["public readonly"] }` options:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> ee12506f5 (figuring out lint)
Examples of **correct** code for the `{ "allows": ["public readonly"] }` options:
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
<<<<<<< HEAD
>>>>>>> 84b8e337c (figuring out lint)
=======
=======
#### ‚úÖ Correct
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> ee12506f5 (figuring out lint)

```ts
class Foo {
  constructor(name: string) {}
}

class Foo {
  constructor(public readonly name: string) {}
}
```

## When Not To Use It

If you don't care about the using parameter properties in constructors, then you will not need this rule.
<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
=======
=======
>>>>>>> 84b8e337c (figuring out lint)
=======
>>>>>>> ee12506f5 (figuring out lint)

## Compatibility

- TSLint: [no-parameter-properties](https://palantir.github.io/tslint/rules/no-parameter-properties/)
<<<<<<< HEAD
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
>>>>>>> 6be43a322 (figuring out lint)
<<<<<<< HEAD
>>>>>>> 84b8e337c (figuring out lint)
=======
=======
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> ee12506f5 (figuring out lint)
