<<<<<<< HEAD
<<<<<<< HEAD
---
description: 'Disallow `require` statements except in import statements.'
---

> ðŸ›‘ This file is source code, not the primary documentation location! ðŸ›‘
>
> See **https://typescript-eslint.io/rules/no-var-requires** for documentation.

In other words, the use of forms such as `var foo = require("foo")` are banned. Instead use ES6 style imports or `import foo = require("foo")` imports.

## Examples

<!--tabs-->

### âŒ Incorrect
=======
=======
>>>>>>> 84b8e337c (figuring out lint)
# Disallows the use of require statements except in import statements (`no-var-requires`)

In other words, the use of forms such as `var foo = require("foo")` are banned. Instead use ES6 style imports or `import foo = require("foo")` imports.

## Rule Details

Examples of **incorrect** code for this rule:
<<<<<<< HEAD
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
---
description: 'Disallow `require` statements except in import statements.'
---

> ðŸ›‘ This file is source code, not the primary documentation location! ðŸ›‘
>
> See **https://typescript-eslint.io/rules/no-var-requires** for documentation.

In other words, the use of forms such as `var foo = require("foo")` are banned. Instead use ES6 style imports or `import foo = require("foo")` imports.

## Examples

<!--tabs-->

### âŒ Incorrect
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> 84b8e337c (figuring out lint)

```ts
var foo = require('foo');
const foo = require('foo');
let foo = require('foo');
```

<<<<<<< HEAD
<<<<<<< HEAD
### âœ… Correct
=======
Examples of **correct** code for this rule:
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
Examples of **correct** code for this rule:
=======
### âœ… Correct
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> 84b8e337c (figuring out lint)

```ts
import foo = require('foo');
require('foo');
import foo from 'foo';
```

## When Not To Use It

<<<<<<< HEAD
<<<<<<< HEAD
If you don't care about using newer module syntax, then you will not need this rule.

## Related To

- [`no-require-imports`](./no-require-imports.md)
=======
=======
>>>>>>> 84b8e337c (figuring out lint)
If you don't care about TypeScript module syntax, then you will not need this rule.

## Compatibility

- TSLint: [no-var-requires](https://palantir.github.io/tslint/rules/no-var-requires/)
<<<<<<< HEAD
>>>>>>> cbaeadb69664be713d49558df22b6f02c48f3384
=======
=======
If you don't care about using newer module syntax, then you will not need this rule.

## Related To

- [`no-require-imports`](./no-require-imports.md)
>>>>>>> 6be43a322 (figuring out lint)
>>>>>>> 84b8e337c (figuring out lint)
